eventstore.go

```diff
@@ -67,6 +69,62 @@ type Event struct {
 	Sequence uint64
 }
 
+type NewEventOption func(*Event) error
+
+func NewEventID(id string) NewEventOption {
+	return func(e *Event) error {
+		if id == "" {
+			return ErrEventIDRequired
+		}
+		e.ID = id
+		return nil
+	}
+}
+
+func NewEventTime(t time.Time) NewEventOption {
+	return func(e *Event) error {
+		if t.IsZero() {
+			return ErrTimeFieldRequired
+		}
+		e.Time = t
+		return nil
+	}
+}
+
+func NewEventType(t string) NewEventOption {
+	return func(e *Event) error {
+		e.Type = t
+		return nil
+	}
+}
+
+func NewEventMetadata(metadata map[string]string) NewEventOption {
+	return func(e *Event) error {
+		if metadata != nil {
+			e.Meta = metadata
+		}
+		return nil
+	}
+}
+
+func (s *EventStore) NewEvent(data any, opts ...NewEventOption) (*Event, error) {
```

Comment:
Given this doesn't depend on `EventStore` at all, this should be a standalone method.